tests/functional/olp-cpp-sdk-authentication/FederatedAuthenticationTest.cpp

```diff
@@ -0,0 +1,210 @@
+  EXPECT_TRUE(response.GetResult().GetAccessToken().empty());
+  EXPECT_TRUE(response.GetResult().GetTokenType().empty());
+  EXPECT_TRUE(response.GetResult().GetRefreshToken().empty());
+  EXPECT_TRUE(response.GetResult().GetUserIdentifier().empty());
+  EXPECT_TRUE(response.GetResult().GetTermAcceptanceToken().empty());
+  EXPECT_TRUE(response.GetResult().GetTermsOfServiceUrl().empty());
+  EXPECT_TRUE(response.GetResult().GetTermsOfServiceUrlJson().empty());
+  EXPECT_TRUE(response.GetResult().GetPrivatePolicyUrl().empty());
+  EXPECT_TRUE(response.GetResult().GetPrivatePolicyUrlJson().empty());
+}
+
+TEST_F(FederatedAuthenticationTest, SignInGoogle) {
+  ASSERT_FALSE(token_.access_token.empty());
+
+  const std::string email = GetEmail();
+  std::cout << "Creating account for: " << email << std::endl;
+
+  AuthenticationClient::SignInUserResponse response =
+      SignInFederated(GoogleAuthenticationBody(email, token_.access_token));
+  EXPECT_EQ(olp::http::HttpStatusCode::CREATED,
+            response.GetResult().GetStatus());
+  EXPECT_EQ(kErrorPreconditionCreatedCode,
+            response.GetResult().GetErrorResponse().code);
+  EXPECT_EQ(kErrorPreconditionCreatedMessage,
+            response.GetResult().GetErrorResponse().message);
+  EXPECT_TRUE(response.GetResult().GetAccessToken().empty());
+  EXPECT_TRUE(response.GetResult().GetTokenType().empty());
+  EXPECT_TRUE(response.GetResult().GetRefreshToken().empty());
+  EXPECT_TRUE(response.GetResult().GetUserIdentifier().empty());
+  EXPECT_FALSE(response.GetResult().GetTermAcceptanceToken().empty());
+  EXPECT_FALSE(response.GetResult().GetTermsOfServiceUrl().empty());
+  EXPECT_FALSE(response.GetResult().GetTermsOfServiceUrlJson().empty());
+  EXPECT_FALSE(response.GetResult().GetPrivatePolicyUrl().empty());
+  EXPECT_FALSE(response.GetResult().GetPrivatePolicyUrlJson().empty());
+
+  std::cout << "termAcceptanceToken="
+            << response.GetResult().GetTermAcceptanceToken() << std::endl;
+
+  AuthenticationClient::SignInUserResponse response2 = AcceptTerms(response);
+  EXPECT_EQ(olp::http::HttpStatusCode::NO_CONTENT,
+            response2.GetResult().GetStatus());
+  EXPECT_STREQ(kErrorNoContent.c_str(),
+               response2.GetResult().GetErrorResponse().message.c_str());
+  EXPECT_TRUE(response2.GetResult().GetAccessToken().empty());
+  EXPECT_TRUE(response2.GetResult().GetTokenType().empty());
+  EXPECT_TRUE(response2.GetResult().GetRefreshToken().empty());
+  EXPECT_TRUE(response2.GetResult().GetUserIdentifier().empty());
+  EXPECT_TRUE(response2.GetResult().GetTermAcceptanceToken().empty());
+  EXPECT_TRUE(response2.GetResult().GetTermsOfServiceUrl().empty());
+  EXPECT_TRUE(response2.GetResult().GetTermsOfServiceUrlJson().empty());
+  EXPECT_TRUE(response2.GetResult().GetPrivatePolicyUrl().empty());
+  EXPECT_TRUE(response2.GetResult().GetPrivatePolicyUrlJson().empty());
+
+  AuthenticationClient::SignInUserResponse response3 =
```

Comment:
I would put all under different scopes with scope trace to at least know what you are actually testing.